Update to poetry

blakesweeney opened the pull request:
This now uses poetry to build the env and track dependencies. As I updated several packages I made to make a small change to the validator to match the new version of jsonschema. Everything still seems to work.

blakesweeney committed:
Update to poetry

This now uses poetry to build the env and track dependencies. As I
updated several packages I made to make a small change to the validator
to match the new version of jsonschema. Everything still seems to work.

afg1 reviewed (commented):
All looks good to me.

Would you like to add poetry to the `pre-commit` hooks? If so, I would reccomend doing it how it is in the import pipeline dev branch. Theres an argument to not update the lockfile on every commit that makes life easier. 

Happy to add that myself if you like

blakesweeney committed:
Add poetry pre-commit hooks

Just copies from RNAcentral pipeline to update the lock file and check things
are valid.

blakesweeney commented:
Added the pre-commit hooks, thanks for the suggestion!
